src/main/java/com/minecolonies/blockout/util/Parsing.java

```diff
@@ -0,0 +1,170 @@
+    {
+        try
+        {
+            int value = Integer.parseInt(m.group(1));
+
+            if ("%".equals(m.group(2)))
+            {
+                value = scale * MathHelper.clamp(value, 0, 100) / 100;
+            }
+            //  DO NOT attempt to do a "value < 0" treated as (100% of parent) - abs(size)
+            //  without differentiating between 'size' and 'position' value types
+            //  even then, it's probably not actually necessary...
+
+            return value;
+        }
+        catch (NumberFormatException | IndexOutOfBoundsException | IllegalStateException ex)
+        {
+            getLogger().warn(ex);
+        }
+
+        return def;
+    }
+
+    /**
+     * Parse the size pair attribute.
+     *
+     * @param attr  the name.
+     * @param def   the definition.
+     * @param scale the scale.
+     * @return the SizePair.
+     */
+    @Nullable
+    public static SizePair parseSizePairAttribute(final String attr, final SizePair def, final SizePair scale)
+    {
+        if (attr != null)
+        {
+            int w = def != null ? def.getX() : 0;
+            int h = def != null ? def.getY() : 0;
+
+            final Matcher m = PERCENTAGE_PATTERN.matcher(attr);
+            if (m.find())
+            {
+                w = Parsing.parseScalableIntegerUsingMatcher(m, w, scale != null ? scale.getX() : 0);
+
+                if (m.find() || m.find(0))
+                {
+                    //  If no second value is passed, use the first value
+                    h = Parsing.parseScalableIntegerUsingMatcher(m, h, scale != null ? scale.getY() : 0);
+                }
+            }
+
+            return new SizePair(w, h);
+        }
+
+        return def;
+    }
+
+    /**
+     * Parse the color attribute from name and definition.
+     *
+     * @param attr the attr.
+     * @param def  the definition
+     * @return int color value.
+     */
+    public static int parseColor(final String attr, final int def)
+    {
+        if (attr == null)
+        {
+            return def;
+        }
+
+        final Matcher m = RGBA_PATTERN.matcher(attr);
+
+        if (attr.startsWith("#"))
+        {
+            //  CSS Hex format: #00112233
+            return Integer.parseInt(attr.substring(1), 16);
+        }
+        //  CSS RGB format: rgb(255,0,0) and rgba(255,0,0,0.3)
+        else if ((attr.startsWith("rgb(") || attr.startsWith("rgba(")) && m.find())
+        {
+            return getRGBA(attr, m);
+        }
+        else
+        {
+            return getColorByNumberOrName(def, attr);
+        }
+    }
+
+    private static int getRGBA(final String attr, final Matcher m)
+    {
+        final int r = MathHelper.clamp(Integer.parseInt(m.group(1)), 0, 255);
+        final int g = MathHelper.clamp(Integer.parseInt(m.group(2)), 0, 255);
+        final int b = MathHelper.clamp(Integer.parseInt(m.group(3)), 0, 255);
+
+        int color = ((r & 0xFF) << 16) | ((g & 0xFF) << 8) | (b & 0xFF);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 8 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)